hoptimator-operator/src/main/java/com/linkedin/hoptimator/operator/subscription/SubscriptionReconciler.java

```diff
@@ -106,6 +150,102 @@ Pipeline pipeline(V1alpha1Subscription object) throws Exception {
     return impl.pipeline(sink);
   }
 
+  private boolean apply(String yaml, V1alpha1Subscription owner) {
+    V1OwnerReference ownerReference = new V1OwnerReference();
+    ownerReference.kind(owner.getKind());
+    ownerReference.name(owner.getMetadata().getName());
+    ownerReference.apiVersion(owner.getApiVersion());
+    ownerReference.uid(owner.getMetadata().getUid());
+
+    DynamicKubernetesObject obj = Dynamics.newFromYaml(yaml);
+    String namespace = obj.getMetadata().getNamespace();
+    String name = obj.getMetadata().getName();
+    KubernetesApiResponse<DynamicKubernetesObject> existing = operator.apiFor(obj).get(namespace, name);
+    if (existing.isSuccess()) {
+      String resourceVersion = existing.getObject().getMetadata().getResourceVersion();
+      log.info("Updating existing downstream resource {}/{} {} as \n{}",
+        namespace, name, resourceVersion, yaml);
+      List<V1OwnerReference> owners = existing.getObject().getMetadata().getOwnerReferences();
+      if (owners == null) {
+        owners = new ArrayList<>();
+      }
+      if (owners.stream().anyMatch(x -> x.getUid().equals(ownerReference.getUid()))) {
+        log.info("Existing downstream resource {}/{} is already owned by {}/{}.",
+          namespace, name, ownerReference.getKind(), ownerReference.getName());
```

Comment:
If the owners are matched, is the subsequent API call still necessary? I was wondering if it is reasonable to `return true` here. If not here, should there be a test to see if the update is necessary, ie. `allMatch()` vs `anyMatch()`?

Reply:
The api call is still necessary, even if the ownership doesn't change, cuz anything else in the spec may change.